### PR TITLE
DataReader, DataSupplier implementetion (fixes #1)

### LINKED ISSUE #1: Implementation of DataReader, JsonReader, DataSuppliers
Need to add into project:
- interface DataReader and JsonReader class using SPI technique ;
- data supplier;
- annotation Data.

### PR DESCRIPTION
**Pull request by emilshina**
- implemented DataReader using SPI;
- added DataSupplier for reading data from json files.

### CODE REVIEW

File: src/main/java/io/github/sskorol/core/ReaderListener.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+import static io.github.sskorol.utils.ServiceLoaderUtils.load;
+import static java.lang.ClassLoader.getSystemClassLoader;
+import static org.apache.commons.io.FilenameUtils.getExtension;
+
+/**
+ * Listener for each suite.
+ */
+
+@Slf4j
```
Comment (mkgerasimenko) [inline src/main/java/io/github/sskorol/core/ReaderListener.java]: need to delete unused annotations

File: src/test/java/io/github/sskorol/data/DataSuppliers.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import java.lang.reflect.Method;
+
+import static io.github.sskorol.core.ReaderListener.readerOf;
+import static java.util.Optional.ofNullable;
+
+/**
+ * Data suppliers' methods for tests.
+ */
+
+@Slf4j
```
Comment (mkgerasimenko) [inline src/test/java/io/github/sskorol/data/DataSuppliers.java]: need to delete unused annotations

File: src/main/java/io/github/sskorol/core/ReaderListener.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static io.github.sskorol.utils.ServiceLoaderUtils.load;
+import static java.lang.ClassLoader.getSystemClassLoader;
+import static org.apache.commons.io.FilenameUtils.getExtension;
+
+/**
+ * Listener for each suite.
```
Comment (mkgerasimenko) [inline src/main/java/io/github/sskorol/core/ReaderListener.java]: add a more obvious description for this class.

File: src/test/resources/META-INF/services/io.github.sskorol.data.DataReader [hunk 4]
```diff
@@ -0,0 +1 @@
+io.github.sskorol.data.JsonReader
```
Comment (lmalakhova) [inline src/test/resources/META-INF/services/io.github.sskorol.data.DataReader]: Add new line.

**Comment by emilshina** [comment]
All comments were fixed.

**Review by mkgerasimenko — APPROVED** [review]

**Review by lmalakhova — APPROVED** [review]

**Pull request merged**